OpenRA.Mods.Common/UtilityCommands/ExtractWeaponDocsCommand.cs

```diff
@@ -0,0 +1,97 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2016 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using OpenRA.GameRules;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.UtilityCommands
+{
+	class ExtractWeaponDocsCommand : IUtilityCommand
+	{
+		string IUtilityCommand.Name { get { return "--weapon-docs"; } }
+
+		bool IUtilityCommand.ValidateArguments(string[] args)
+		{
+			return true;
+		}
+
+		[Desc("Generate weaponry documentation in MarkDown format.")]
+		void IUtilityCommand.Run(Utility utility, string[] args)
+		{
+			// HACK: The engine code assumes that Game.modData is set.
+			Game.ModData = utility.ModData;
+
+			Console.WriteLine(
+				"This documentation is aimed at modders. It displays a template for weapon definitions " +
+				"as well as its contained types (warheads and projectiles) with default values and developer commentary. " +
+				"Please do not edit it directly, but add new `[Desc(\"String\")]` tags to the source code. This file has been " +
+				"automatically generated for version {0} of OpenRA.", utility.ModData.Manifest.Metadata.Version);
+			Console.WriteLine();
+
+			var toc = new StringBuilder();
+			var doc = new StringBuilder();
+			var currentNamespace = "";
+
+			var weaponInfo = Game.ModData.ObjectCreator.GetTypesImplementing<WeaponInfo>();
```

Comment:
Style nit: you should use `utility.ModData` instead of `Game.ModData`, and could pull `utility.ModData.ObjectCreator` to its own temporary variable.